### DISCUSSION TITLE
Integration Using SDK/Library

### DISCUSSION THREAD

**Discussion by Ronnie-personal** [open-policy-agent/community#465] [OPA and Rego] [answered]
Hi,
Based on the document provided at https://www.openpolicyagent.org/docs/latest/integration/#comparison, it states that OPA currently only has a Go SDK.

Is there a roadmap or any plans in place to support a native Java SDK?
Thanks,
Ronnie

**Comment by charlieegan3** [comment, marked as answer]
Hi Ronnie, this isn't something that we have planned. We're able to offer the Go SDK relatively easily since OPA itself is built in Go.

There are however some community Java integrations,

* https://github.com/Bisnode/opa-java-client for talking to OPA over the REST API
* https://github.com/sangkeon/java-opa-wasm for running OPA Wasm modules within Java applications

Have a look in the [awesome-opa](https://github.com/StyraInc/awesome-opa#java) list as there are also a few other projects that might be good for your use case.

**Reply by Ronnie-personal** [reply]
Hi @charlieegan3 

Thank you for the information.

I'm looking for an option to evaluate policy right inside the code stack without running a separate OPA server. WebAssembly (Wasm) could be an option, but I need to understand how to make the http.send function work. This project, https://github.com/borgeby/jarl, looks interesting too

**Reply by charlieegan3** [reply]
http.send in Wasm would need the SDK to support it as an external built-in. It looks like there is some mechanism to support adding external built-ins but I'm not enough of a Java person to work out how it'd be used: https://github.com/search?q=repo%3Asangkeon%2Fjava-opa-wasm%20builtinsFn&type=code.

This is how additional built-ins are added in the NPM OPA Wasm module: https://github.com/open-policy-agent/npm-opa-wasm/blob/main/src/builtins/strings.js
